cg/services/order_validation_service/models/order_with_cases.py

```diff
@@ -1,9 +1,19 @@
+from pydantic import Discriminator, Tag
+from typing_extensions import Annotated
+
 from cg.services.order_validation_service.models.case import Case
+from cg.services.order_validation_service.models.discriminators import has_internal_id
+from cg.services.order_validation_service.models.existing_case import ExistingCase
 from cg.services.order_validation_service.models.order import Order
 
 
 class OrderWithCases(Order):
-    cases: list[Case]
+    cases: list[
+        Annotated[
+            Annotated[Case, Tag("new")] | Annotated[ExistingCase, Tag("existing")],
+            Discriminator(has_internal_id),
+        ]
+    ]
```

Comment:
Ugly typing - should I break it up?

Reply:
I think it is fine, if you prefer something like this go for it
```python
NewCaseType: TypeAlias = Annotated[Case, Tag("new")]
ExistingCaseType: TypeAlias = Annotated[ExistingCase, Tag("existing")]
DiscriminatedCaseType: TypeAlias = Annotated[Union[NewCaseType, ExistingCaseType], Discriminator(has_internal_id)]

cases: List[DiscriminatedCaseType]
```

Reply:
I'll go for a hybrid! I like `NewCaseType` and `ExistingCaseType` but it might make sense to see that there is a union of the two classes still